src/sinks/gcp/stackdriver/metrics/mod.rs

```diff
@@ -0,0 +1,20 @@
+// TODO: In order to correctly assert component specification compliance, we would have to do some more advanced mocking
+// off the endpoint, which would include also providing a mock OAuth2 endpoint to allow for generating a token from the
+// mocked credentials. Let this TODO serve as a placeholder for doing that in the future.
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Wondering if this might pose implications for the Component Validation Framework rollout to this sink. We have a concept of "external resources" which are there to serve the purpose of the mocking of the endpoint, but they are generic and don't consider any `OAuth` things.

Reply:
Ah, this comment no longer applies. We can mock the endpoint with configuration like this:

```
 let config = StackdriverConfig {
        endpoint: format!("http://{in_addr}"),
        auth: GcpAuthConfig {
            api_key: None,
            credentials_path: None,
            skip_authentication: true,
        },
        ..Default::default()
    };
```

It's done in the tests already.

I'll remove the comment.